src/Templates/Components/modals/modal-new-pwd.pug

```diff
@@ -0,0 +1,242 @@
+      ILLEGAL_NOT_PERSONAL: 'Kun personlige logins kan ændre kodeord'
+    }
+    var infoStep1 = 'Udfyld Biblioteksnummer og Bruger ID og vælg Mail sikkerhedskode. Du får en mail med en sikkerhedskode, som skal bruges derefter.'
+    var infoStep2 = 'Udfyld sikkerhedskode, som du finder i den mail, der netop er sendt til dig og vælg en ny sikker adgangskode.'
+    var activeErrorField = '';
+    var identityNewPasswordInput = document.getElementById('identityNewPassword');
+    var agencyIdNewPasswordInput = document.getElementById('agencyIdNewPassword');
+    var twoFactorCodeInput = document.getElementById('twoFactorCode');
+    var newPasswordInput = document.getElementById('newPassword');
+    var checkPasswordInput = document.getElementById('checkPassword');
+    var messageContainerNewPass = document.getElementById('new-pw-message-container');
+    var responseMessageNewPass = document.getElementById('new-pw-response-message');
+    var closeButton = document.querySelector('.newPw-close-button');
+    var submitStep1 = document.getElementById('newPw-submit-step1');
+    var submitStep2 = document.getElementById('newPw-submit-step2');
+    var submitCodeContainer = document.getElementById('newPw-code-container')
+    var submitCodePassContainer = document.getElementById('newPw-code-pass-container')
+    var secretHash = Date.now();
+
+    if (closeButton) {
+      closeButton.addEventListener('click', function(e) {
+        messageContainerNewPass.style.display = "none";
+      });
+    }
+
+    responseMessageNewPass.textContent = infoStep1;
+    messageContainerNewPass.style.display = 'block';
+
+    document.getElementById('newpw-form').addEventListener('submit', function(e) {
+      e.preventDefault();
+      if (activeErrorField) {
+        document.getElementById(activeErrorField).textContent = '';
+        activeErrorField = '';
+      }
+      if (submitStep2.value == 'submitStep2') {
+        if (newPwSubmitStep2(e)) {
+          var data = {
+            'identity': identityNewPasswordInput.value,
+            'agencyid': agencyIdNewPasswordInput.value,
+            'hash': secretHash,
+            'secret': twoFactorCodeInput.value,
+            'password': newPasswordInput.value,
+          };
+          fetch('/newpasswordstep2', {
+            method: 'POST',
+            headers: {
+              'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(data)
+          })
+          .then(response => response.json())
+          .then(data => {
+            if (!data.valid || data.message !== 'PASSWORD_CHANGED') {
+              var errorMessage = data.message.split(',');
+              let passwordMsg = [];
+              errorMessage.forEach((err) => {
+                if (passwordErrorText[err]) passwordMsg.push(passwordErrorText[err]);
+              });
+              let identityMsg = [];
+              errorMessage.forEach((err) => {
```

Comment:
Du looper errorMessage igennem to gange?
Kan det ikke gøres smartere?

Reply:
Jov. Done